submit_async.go

```diff
@@ -0,0 +1,79 @@
+package gosnowflake
+
+// this is a sigma only file
+// when rebasing you can ignore everthing in this file as long as this
+// file makes it into master
+
+import (
+	"context"
+	"database/sql/driver"
+	"strconv"
+)
+
+type isComplete bool
+type queryID string
+
+func (sc *snowflakeConn) SubmitAsync(
+	ctx context.Context,
+	query string,
+	args []driver.NamedValue,
+) (queryID, isComplete, error) {
+	ctx = WithAsyncMode(WithAsyncModeNoFetch(ctx))
+	qid, err := getResumeQueryID(ctx)
+	if err != nil {
+		return "", false, err
+	}
+
+	// if query already submitted, return the query id
+	if qid != "" {
+		return queryID(qid), false, nil
+	}
+
+	// if the query is not submitted, submit it
+	logger.WithContext(ctx).Infof("Query: %#v, %v", query, args)
+
+	if sc.rest == nil {
+		return "", false, driver.ErrBadConn
```

Comment:
I'd suggest returning a different error type. `driver.ErrBadConn` is a special error that `database/sql` uses to detect when to cleanup a connection that's in a bad state. But this call doesn't go through `database/sql`, so it won't have that effect.

Reply:
I was planning on calling it like this 
```
err = conn.Raw(
        func(x interface{}) error {
            queryId, complete, _ := x.(snowflake.AsyncSubmitter).SubmitAsync(ctx, query, args)
      ... 
```
and my thought process was that since this connection was established by dbsql, who creates the `snowflakeConn` and populates it with the `snowflakeRestful` that there could be a chance that rest isnt populated if dbSql did something in the `conn, err := a.getConn(ctx)` step. But maybe there is a better way to do this